### PR TITLE
Add x-goog-user-proj header for sts credential

### PR DESCRIPTION
**Pull request by bianpengyuan**
Signed-off-by: Pengyuan Bian <bianpengyuan@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: api/wasm/cpp/proxy_wasm_api.h [hunk 1]
```diff
@@ -1234,19 +1236,17 @@ inline void GrpcStreamHandlerBase::send(StringView message, bool end_of_stream)
   }
 }
 
-inline void RootContext::onGrpcCreateInitialMetadata(uint32_t token) {
+inline void RootContext::onGrpcCreateInitialMetadata(uint32_t) {
+  std::cout<<"called ongrpccreateinitialmetadata on root context\n";
```
Comment (mandarjog) [inline api/wasm/cpp/proxy_wasm_api.h]: Probably inadvertent

File: source/extensions/tracers/opencensus/opencensus_tracer_impl.cc [hunk 2]
```diff
@@ -232,6 +232,39 @@ Tracing::SpanPtr Span::spawnChild(const Tracing::Config& /*config*/, const std::
 
 void Span::setSampled(bool sampled) { span_.AddAnnotation("setSampled", {{"sampled", sampled}}); }
 
+class GoogleUserProjHeaderInterceptor : public grpc::experimental::Interceptor {
+public:
+  GoogleUserProjHeaderInterceptor(const std::string& project_id) : project_id_(project_id) {}
+
+  virtual void Intercept(grpc::experimental::InterceptorBatchMethods* methods) {
+    if (methods->QueryInterceptionHookPoint(
```
Comment (mandarjog) [inline source/extensions/tracers/opencensus/opencensus_tracer_impl.cc]: @g-easy does this look ok?
It is being directly pushed into an intermediary version of envoy in istio.

File: api/wasm/cpp/proxy_wasm_api.h (hunk 1 above)
Comment (bianpengyuan) [reply api/wasm/cpp/proxy_wasm_api.h]: Done.

File: source/extensions/tracers/opencensus/opencensus_tracer_impl.cc (hunk 2 above)
Comment (bianpengyuan) [reply source/extensions/tracers/opencensus/opencensus_tracer_impl.cc]: This should not be needed in envoy master after https://github.com/grpc/grpc/issues/21225 is resolved.

Comment (g-easy) [reply source/extensions/tracers/opencensus/opencensus_tracer_impl.cc]: This seems reasonable to me but I haven't dealt with interceptors before.

**Comment by kyessenov** [comment]
Xref: https://github.com/envoyproxy/envoy-wasm/issues/414
CC @PiotrSikora 
This looks fine to me, please take another look.

**Comment by bianpengyuan** [comment]
Let me separate the wasm change into a upstream PR.

**Pull request merged**